crates/macro-impl/src/html_tree/html_component.rs

```diff
@@ -48,35 +48,56 @@ impl Parse for HtmlComponent {
 impl ToTokens for HtmlComponent {
     fn to_tokens(&self, tokens: &mut proc_macro2::TokenStream) {
         let HtmlComponentInner { ty, props } = &self.0;
-        let vcomp_props = Ident::new("__yew_vcomp_props", Span::call_site());
         let vcomp_scope = Ident::new("__yew_vcomp_scope", Span::call_site());
-        let override_props = props.iter().map(|props| match props {
-            Props::List(ListProps(vec_props)) => {
-                let check_props = vec_props.iter().map(|HtmlProp { label, .. }| {
-                    quote_spanned! { label.span()=> #vcomp_props.#label; }
-                });
-
-                let set_props = vec_props.iter().map(|HtmlProp { label, value }| {
-                    quote_spanned! { value.span()=>
-                        #vcomp_props.#label = <::yew::virtual_dom::vcomp::VComp<_> as ::yew::virtual_dom::vcomp::Transformer<_, _, _>>::transform(#vcomp_scope.clone(), #value);
-                    }
-                });
 
-                quote! {
-                    #(#check_props#set_props)*
+        let validate_props = if let Some(Props::List(ListProps(vec_props))) = props {
+            let prop_ref = Ident::new("__yew_prop_ref", Span::call_site());
+            let check_props = vec_props.iter().map(|HtmlProp { label, .. }| {
+                quote! { #prop_ref.#label; }
+            });
+
+            quote! {
```

Comment:
This purpose of this block is to validate that all props passed in `<Component prop=0 ... />` are actually present on the components properties struct. If not, a nice compile error message will be shown. The unfortunate side effect is that it has a runtime performance hit because of the `zeroed` call. Still searching for a better solution here..